haicrypt/hcrypt.c

```diff
@@ -39,63 +39,105 @@ int HaiCrypt_SetLogLevel(int level, int logfa)
     return 0;
 }
 
+#if ENABLE_HAICRYPT_LOGGING
 
-int HaiCrypt_Create(const HaiCrypt_Cfg *cfg, HaiCrypt_Handle *phhc)
+static const char* DumpCfgFlags(int flags)
+{
+    static char buf[4096];
```

Comment:
Please avoid returning non-const static buffers from a function. There is really no need for this: just pass destination buffer and its length as arguments.